app/styles/molecules/social-media-badge.less

```diff
@@ -0,0 +1,171 @@
+.generate-social-media-badge(@social-media) {
```

Comment:
:/ 
We are mostly trying to get rid of using less, and although we still use it sometimes, I feel here is not a justified usage.
You already define css classes for the social media types (for example line 85/88) I think all of the common attributes should be in a common class that is always defined.

Rename `.generate-social-media-badge` to `social-media-badge`, give it default values for `background-color`, `border-color` and the .iconcontainer's `background`.
Then overwrite these three values in your social-media-types definition, something like : 
```css
  &--twitter {
    background-color: #color for twitter
    border-color: #color for twitter

    .icon-container {
        background: #background for twitter
    }
  }
```

Your computedClasses getter should already be working with this solution.